### PR TITLE
Bugfix/flow keyword set error - v2

### PR DESCRIPTION
**Pull request by open-cmi**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/codebase/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- fix the flag test
- write an unit test
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Comment by open-cmi** [comment]
What else do I need to do ? I want to merge the pr  first and start to fix bug according to next ticket.

**Review by catenacyber — CHANGES_REQUESTED** [review]
Thanks for the work :-)

- CI : expecting it to be red because of `not_established,established` case not being handled
- Code : need to handle the other way around `not_established,established`
- Commits segmentation : Could you squash both commits into one ?
- Commit messages : Still need to add the reference to the redmine ticket number
- Git ID set : looks fine for me
- CLA : I do not have access to it
- Doc update : None needed
- Redmine ticket : ok
- Rustfmt : none needed
- Tests : unit test looks fine
- Dependencies added: none

**Comment by open-cmi** [comment]
> Thanks for the work :-)
> 
> * CI : expecting it to be red because of `not_established,established` case not being handled
> * Code : need to handle the other way around `not_established,established`
> * Commits segmentation : Could you squash both commits into one ?
> * Commit messages : Still need to add the reference to the redmine ticket number
> * Git ID set : looks fine for me
> * CLA : I do not have access to it
> * Doc update : None needed
> * Redmine ticket : ok
> * Rustfmt : none needed
> * Tests : unit test looks fine
> * Dependencies added: none

Thank you very much

File: src/detect-flow.c [hunk 1]
```diff
@@ -227,6 +227,9 @@ static DetectFlowData *DetectFlowParse (DetectEngineCtx *de_ctx, const char *flo
                 if (fd->flags & DETECT_FLOW_FLAG_ESTABLISHED) {
                     SCLogError("DETECT_FLOW_FLAG_ESTABLISHED flag is already set");
                     goto error;
+                } else if (fd->flags & DETECT_FLOW_FLAG_NOT_ESTABLISHED) {
+                    SCLogError("DETECT_FLOW_FLAG_NOT_ESTABLISHED flag is already set");
```
Comment (catenacyber) [inline src/detect-flow.c:231]: error message should be different, right ?

Comment (open-cmi) [reply src/detect-flow.c]: Thank you, I have changed the error message. Is it OK now?

**Comment by catenacyber** [comment]
For the commit message,  I think we need an empty line after title

**Comment by open-cmi** [comment]
> For the commit message, I think we need an empty line after title

OK.

**Comment by victorjulien** [comment]
Setting to draft as the PR history is messy, and we'll want a fresh PR before merge as per our contributing docs.

**Comment by jufajardini** [comment]
I think that the CI failures are due to lack of a rebase. Could you please open a new PR with your code rebased, so we can have a possible merge candidate for your contribution? :)

**Comment by open-cmi** [comment]
> I think that the CI failures are due to lack of a rebase. Could you please open a new PR with your code rebased, so we can have a possible merge candidate for your contribution? :)

I rebased on my branch, Is that OK? Or need I create a new branch and open a new PR?

**Comment by jufajardini** [comment]
> > I think that the CI failures are due to lack of a rebase. Could you please open a new PR with your code rebased, so we can have a possible merge candidate for your contribution? :)
> 
> I rebased on my branch, Is that OK? Or need I create a new branch and open a new PR?

Thanks for asking. As Victor pointed out, we'll want a fresh PR before merge, so, yes, we do need a new PR. ;)

**Comment by open-cmi** [comment]
OK, I open a new PR at #9783

**Comment by jufajardini** [comment]
Replaced by: https://github.com/OISF/suricata/pull/9783

**Pull request closed without merging**